build.sh

```diff
@@ -121,49 +154,138 @@ function testModels() {
-CERTIFAI_VERSION=$(getToolkitVersion)
-GIT_SHA=$(git log -1 --pretty=%h)
-VERSION="v4-${CERTIFAI_VERSION}"
-echo "##### BUILDING ${VERSION} ######"
-case ${1-local} in
- CI)
-  activateConda
-  installToolkit
-  test
-  rm -rf ${TOOLKIT_WORK_DIR}
-  ;;
- docker)
-  PUSH_IMAGES=true
-  build_model_deployment_base_images
-  ;;
- *)
-  buildLocal
-  ;;
-esac
+function _runNotebookInPlace() {
+  # FYI - stdout/stderr from the notebook is NOT redirected by nbcovert (if needed check the log file at "~/.certifai")
+  jupyter nbconvert --to notebook --inplace --execute $1
+}
+
+# Examples involving multiple notebooks explicit order
+MULTIPART_NOTEBOOKS=(patient_readmission data_statistics)
+
+# Examples requiring a new conda env or new dependencies
+NOTEBOOKS_REQUIRING_ENV_SETUP=(azureml_model_headers_demo sagemaker target_encoded xgboost-model)
+
+# Example notebook folders to skip (usually empty). Useful in avoiding
+# recomputes when a notebook bombs whilst running the script.
+EXCLUDE_SINGULAR_NOTEBOOK=()
+
+function runMultipartNotebooks() {
+  # data_statistics
+  _runNotebookInPlace "${NOTEBOOK_DIR}/data_statistics/prep_adult_data_drift.ipynb"
+  _runNotebookInPlace "${NOTEBOOK_DIR}/data_statistics/adult_drift_data_statistics.ipynb"
+
+  # patient_readmission
+  _runNotebookInPlace "${NOTEBOOK_DIR}/patient_readmission/patient-readmission-train.ipynb"
+  _runNotebookInPlace "${NOTEBOOK_DIR}/patient_readmission/patient-readmission-explain-scan.ipynb"
+  _runNotebookInPlace "${NOTEBOOK_DIR}/patient_readmission/patient-readmission-explain-results.ipynb"
+  _runNotebookInPlace "${NOTEBOOK_DIR}/patient_readmission/patient-readmission-trust-scan.ipynb"
+  _runNotebookInPlace "${NOTEBOOK_DIR}/patient_readmission/patient-readmission-trust-results.ipynb"
+  _runNotebookInPlace "${NOTEBOOK_DIR}/patient_readmission/patient-readmission-sampling-scan.ipynb"
+  _runNotebookInPlace "${NOTEBOOK_DIR}/patient_readmission/patient-readmission-sampling-results.ipynb"
+}
+
+function runIndependentNotebooks() {
+  local multipart_grep
+  multipart_grep="${MULTIPART_NOTEBOOKS[*]/#/-e }"
+
+  local env_grep
+  env_grep="${NOTEBOOKS_REQUIRING_ENV_SETUP[*]/#/-e }"
+
+  local exclude_grep
+  exclude_grep="${EXCLUDE_SINGULAR_NOTEBOOK[*]/#/-e }"
+
+  # All examples with independent notebooks
+  local notebooks
+  # shellcheck disable=SC2086
+  notebooks=$(ls "${NOTEBOOK_DIR}" | grep -v -e README -e datasets -e definitions -e utils ${multipart_grep} ${env_grep} ${exclude_grep})
+
+  # shellcheck disable=SC2068
+  for n in ${notebooks[@]}; do
+    _runNotebookInPlace "${n}/*.ipynb"
+  done
+}
+
+function runNotebooksWithEnvSetup() {
+  _xgboostModel
+  _targetEncodedAzuremlNotebook
+  _azuremlModelHeadersDemo
+  _sagemakerNotebook
+}
+
+function _azuremlModelHeadersDemo() {
+  # TODO: azureml_model_headers_demo
+  true
+}
+
+function _targetEncodedAzuremlNotebook() {
+  # TODO: target_encoded
+  conda remove -n certifai-azure-model-env --all -y
```

Comment:
curious why we are removing this env in this bash func

Reply:
The reason why is that this example has it's conda environment (named `certifai-azure-model-env`), so when re-running the example we should clean/recreate the env.

**This example is not automated in this PR, so this is just some extra/filler code for the moment**.  I've split off the execution of this notebook (and others using azureml/sagemaker) to separate PRs. See: #131 (and #130, #132)